File: math/1513_number_of_substrings_with_only_1s/1513_number_of_substrings_with_only_1s.go
package main

import "fmt"

const mod = 1000000007

func numSub(s string) int {
	sum, count := 0, 0
	for i := 0; i < len(s); i++ {
		if s[i] == '1' {
			count++
		} else {
			sum = (sum + (count+1)*count/2) % mod
			count = 0
		}
	}
	if count != 0 {
		sum = (sum + (count+1)*count/2) % mod
	}
	return sum
}

func main() {
	s := "0110111"
	res := numSub(s)
	fmt.Println(res)
}

File: math/1513_number_of_substrings_with_only_1s/1513_number_of_substrings_with_only_1s.py
class Solution:
    def numSub(self, s: str) -> int:
        cnt = 0
        res = 0
        for i in s:
            if i == '1':
                cnt += 1
            else:
                res += cnt * (cnt +1) /2
                cnt = 0
        if cnt > 0:
            res += cnt * (cnt + 1)/ 2
        res %= 1e9+7
        return int(res)
                
                

s = "0110111"
res = Solution().numSub(s)
print(res)